src/ngraph/node.cpp

```diff
@@ -406,23 +406,26 @@ const NodeVector& ngraph::check_single_output_args(const NodeVector& args)
 
 void Node::validate_and_infer_elementwise(element::Type result_type)
 {
-    const element::Type& element_type = get_input_element_type(0);
-    const Shape& shape = get_input_shape(0);
+    element::Type element_type = get_input_element_type(0);
+    PartialShape pshape = get_input_partial_shape(0);
+
     if (get_input_size() > 1)
     {
         for (size_t i = 1; i < get_input_size(); ++i)
         {
-            NODE_VALIDATION_ASSERT(this, get_input_element_type(i) == element_type)
-                << "Argument 0 element type " << element_type
-                << " differs in element type from argument " << i << " " << *get_argument(i)
-                << " element type " << get_input_element_type(i);
-
-            NODE_VALIDATION_ASSERT(this, get_input_shape(i) == shape)
-                << "Argument 0 shape " << shape << " differs in shape from argument " << i << " "
-                << *get_argument(i) << " shape " << get_input_shape(i);
+            // TODO(amprocte): we need more info on what the arg shapes/types are. Perhaps this
+            // should be produced by NODE_VALIDATION_ASSERT itself.
+
+            NODE_VALIDATION_ASSERT(
+                this, element::Type::merge(element_type, element_type, get_input_element_type(i)))
+                << "Argument element types are inconsistent.";
```

Comment:
There's an unfortunate step back here in terms of error message quality... the issue, basically, is that consistency of partial shapes is not transitive, so we can still have inconsistency even when we can't point to any one pair of arguments whose pshapes contradict each other. I'll think about what to do here and maybe push an update to this PR.

Reply:
I think showing the two shapes in the message solves the problem.

Reply:
I addressed this by adding element type information to `NODE_VALIDATION_ASSERT`'s message (indirectly, by adding it to the text of long node descriptions).